File: scripts/lib/atproto/clients/identity.py
from __future__ import annotations

from typing import Optional

from ..transport import TransportLayer


class IdentityClient:
    def __init__(self, transport: TransportLayer):
        self._t = transport

    def resolve_handle(self, handle: str) -> dict:
        return self._t.get("com.atproto.identity.resolveHandle", {"handle": handle})

    def update_handle(self, handle: str, token: str) -> dict:
        return self._t.post("com.atproto.identity.updateHandle", {"handle": handle}, token=token)

File: scripts/lib/deno/clients/identity.ts
import { TransportLayer } from "../transport.ts";

export class IdentityClient {
  constructor(private transport: TransportLayer) {}

  async resolveHandle(handle: string) {
    return await this.transport.get("com.atproto.identity.resolveHandle", { handle });
  }

  async updateHandle(handle: string, token: string) {
    return await this.transport.post("com.atproto.identity.updateHandle", { handle }, token);
  }
}
